src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -687,6 +688,48 @@ public override ImmutableArray<string> TupleElementNames
         }
 
         /// <summary>
+        /// Get the fields for the tuple's elements (in order and cached).
+        /// </summary>
+        public ImmutableArray<FieldSymbol> TupleElementFields
+        {
+            get
+            {
+                if (_lazyFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyFields, CollectTupleElementFields());
+                }
+
+                return _lazyFields;
+            }
+        }
+
+        private ImmutableArray<FieldSymbol> CollectTupleElementFields()
+        {
+            var builder = ArrayBuilder<FieldSymbol>.GetInstance(_elementTypes.Length, null);
+
+            foreach (var member in GetMembers())
+            {
+                if (member.Kind != SymbolKind.Field)
+                {
+                    continue;
+                }
+
+                int index = (member as TupleFieldSymbol)?.TupleFieldId ??
+                            ((TupleErrorFieldSymbol)member).TupleFieldId;
```

Comment:
Rather than creating your own version of the source, please pull complete implementation of this function from tuples branch. Including relevant APIs added to base classes, there is no need to do all this gymnastics and depend on specific implementation types.